source/MaterialXTest/ShaderGen.cpp

```diff
@@ -21,31 +21,33 @@
 
 namespace mx = MaterialX;
 
-// Observer to add library markers to all nodes. Use the "library" meta-data attribute to do this,
-// until namespace specification is resolved.
-static std::string LIBRARY_ATTRIBUTE("library");
-class LibraryObserver : public mx::Observer
+void loadLibraries(const mx::StringVec& libraryNames, const mx::FilePath& searchPath, mx::DocumentPtr doc)
 {
-public:
-    LibraryObserver() {}
-
-    void onAddElement(mx::ElementPtr /*parent*/, mx::ElementPtr element) override
+    for (const std::string& library : libraryNames)
     {
-        if (_libraryRefName.length())
+        mx::FilePath path = searchPath / library;
+        mx::StringVec filenames;
+        mx::getDocumentsInDirectory(path, filenames);
```

Comment:
Note that we are now loading _everything_ we find in the library directory, so all implementations for all languages. Besides the performance concerns (which I have not looked into yet) this work very well and feels nice and clean both on 
 the library layout side and from the loader side.

Reply:
I wasn't sure but are there any ordering concerns here? i.e. do some files need to be loaded in before others? I don't remember if reading does any reference checks. 

The other possible issue is multiple imports of the same dependent doc. This is a general issue that isn't directly tied to this but I think we want to set the optional flag to "ignore duplicates" on read here for now.

Reply:
There are no reference checks at read-in, so ordering should not be a problem. Duplicate elements however, just like you say, we need to enable the flag for that. I will add that to the LibraryManager on LookdevX side.